### PR TITLE
31 implement basic app security

### PR DESCRIPTION
**Pull request by claudist**
- I have added password security to check for Uppercase letter, lower case letter & 1 number when you register new user.
- I have also updated existing test data passwords to conform to password requirements.

### CODE REVIEW

File: src/main/java/newbank/server/commands/RegisterCommand.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public String execute() {
     } catch (DuplicateCustomerException ex) {
       return "FAIL: " + String.format("Customer name [%s] already exists.", username);
     } catch (PasswordInvalidException e) {
-      return "FAIL: Specified password does not meet the security requirements.";
+      return "FAIL: Password must contain mixture of uppercase and lowercase letters and at least one number.";
```
Comment (jkonan100) [inline src/main/java/newbank/server/commands/RegisterCommand.java:43]: must contain *a* mixture.

Regards,
Jules

File: src/test/java/newbank/server/TestNewBank.java [hunk 2]
```diff
@@ -24,33 +24,33 @@ public static void setupBank() throws DuplicateCustomerException, PasswordInvali
...
   public void throwsWhenRegisteringWithInvalidName()
       throws DuplicateCustomerException, PasswordInvalidException, UsernameInvalidException {
-    bank.addCustomer("12345", "123456");
+    bank.addCustomer("12345", "Abc123");
   }
 
   @Test(expected = PasswordInvalidException.class)
   public void throwsWhenRegisteringWithInvalidPassword() throws DuplicateCustomerException, PasswordInvalidException, UsernameInvalidException {
-	  bank.addCustomer("customer4", "");
+	  bank.addCustomer("customer4", "Abc123");
```
Comment (jkonan100) [inline src/test/java/newbank/server/TestNewBank.java:46]: The password is intentionally wrong in this test.

Kindest Regards,
Jules

File: src/main/java/newbank/server/NewBank.java [hunk 3]
```diff
@@ -75,8 +77,9 @@ private void validateUsername(final String username) throws UsernameInvalidExcep
   // Simple algorithm to check that the password meets the security requirements
   // must be non-empty
   private void validatePassword(final String password) throws PasswordInvalidException {
-    if (password.isEmpty()) {
-      throw new PasswordInvalidException();
+    if (!password.matches("^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).+$")){
+        if (password.isEmpty());
```
Comment (jkonan100) [inline src/main/java/newbank/server/NewBank.java:81]: There is a semicolon after the if statement.

**Review by jkonan100 — CHANGES_REQUESTED** [review]
Close Claudia, a few changes and we should be there.

Kindest Regards,
Jules

**Comment by claudist** [comment]
> Close Claudia, a few changes and we should be there.
> 
> Kindest Regards,
> Jules

Thanks Jules :) only just saw these comments !

File: src/main/java/newbank/server/commands/RegisterCommand.java (hunk 1 above)
Comment (claudist) [reply src/main/java/newbank/server/commands/RegisterCommand.java:43]: will make this change and push now (on branch 89-implement-password-requirements)

**Pull request merged**